src/mlpack/methods/ann/ffn_impl.hpp

```diff
@@ -306,6 +306,15 @@ EvaluateWithGradient(const arma::mat& /* parameters */,
                      const size_t begin,
                      GradType& gradient,
                      const size_t batchSize)
+{}
+template<typename OutputLayerType, typename InitializationRuleType,
+         typename... CustomLayers>
+template<typename eT>
+void FFN<OutputLayerType, InitializationRuleType, CustomLayers...>::Gradient(
```

Comment:
All the work is done by `Gradient()`. So, should I remove `EvaluateWithGradient()`?

Reply:
Not sure what you mean:

https://github.com/mlpack/mlpack/blob/f77ef6197981d04b16dbc3d26fc8c35b4a79bb47/src/mlpack/methods/ann/ffn_impl.hpp#L305-L308

Does more as calling `Gradient`.

Reply:
@zoq actually @sumedhghaisas replaced `EvaluateWithGradient` to `Gradient`  you should take a look at implementation I don't know why?